smithy-model/src/main/java/software/amazon/smithy/model/shapes/EnumShape.java

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *   http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.shapes;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import software.amazon.smithy.model.SourceException;
+import software.amazon.smithy.model.traits.EnumDefaultTrait;
+import software.amazon.smithy.model.traits.EnumDefinition;
+import software.amazon.smithy.model.traits.EnumTrait;
+import software.amazon.smithy.model.traits.EnumValueTrait;
+import software.amazon.smithy.model.traits.Trait;
+import software.amazon.smithy.model.traits.UnitTypeTrait;
+import software.amazon.smithy.model.traits.synthetic.SyntheticEnumTrait;
+import software.amazon.smithy.utils.BuilderRef;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.MapUtils;
+
+public final class EnumShape extends StringShape implements NamedMembers {
+
+    private final Map<String, MemberShape> members;
+    private volatile List<String> memberNames;
+    private volatile Map<String, String> enumValues;
+
+    private EnumShape(Builder builder) {
+        super(builder);
+        members = NamedMemberUtils.computeMixinMembers(
+                builder.getMixins(), builder.members, getId(), getSourceLocation());
+        validateMemberShapeIds();
+        if (members.size() < 1) {
+            throw new SourceException("enum shapes must have at least one member", getSourceLocation());
+        }
+    }
+
+    /**
+     * Gets a map of enum member names to their corresponding values.
+     *
+     * @return A map of member names to enum values.
+     */
+    public Map<String, String> getEnumValues() {
+        if (enumValues == null) {
+            Map<String, String> values = new LinkedHashMap<>(members.size());
+            for (MemberShape member : members()) {
+                if (member.hasTrait(EnumDefaultTrait.ID)) {
+                    values.put(member.getMemberName(), "");
+                    continue;
+                }
+                values.put(member.getMemberName(), member.expectTrait(EnumValueTrait.class).expectStringValue());
+            }
+            enumValues = MapUtils.orderedCopyOf(values);
```

Comment:
nit you don't need to copy here since the map doesn't escape this method. Just a `Collections.unmodifiableMap` works.